src/ServiceControl.Monitoring/QueueLength/QueueLengthDataStore.cs

```diff
@@ -38,12 +77,30 @@ void UpdateSends(IEnumerable<JToken> sends)
                 var key = tags.GetTagValue("key");
                 var value = send.Value<long>("Count");
 
-                calculator.UpdateSentSequence(key, value);
+                var updatedVirtualQueues = calculator.UpdateSentSequence(key, value);
+
+                allUpdatedVirtualQueues.AddRange(updatedVirtualQueues);
             }
+
+            return allUpdatedVirtualQueues.Select(v => v.EndpointName).Distinct().ToList();
+        }
+
+        void SnapshotValueForEndpoint(string endpointName)
+        {
+            var endpointQueueLength = calculator
+                .GetQueueLengths()
```

Comment:
Would it make more sense to call this once and then pass the result around to snapshot each individual endpoint?